src/gt4py/next/ffront/experimental.py

```diff
@@ -12,30 +12,23 @@
 #
 # SPDX-License-Identifier: GPL-3.0-or-later
 
-from dataclasses import dataclass
+import numpy as np
 
-from gt4py.next.type_system import type_specifications as ts
+from gt4py.next import common
+from gt4py.next.ffront.fbuiltins import BuiltInFunction, FieldOffset
 
 
-@dataclass
-class BuiltInFunction:
-    __gt_type: ts.FunctionType
+@BuiltInFunction
+def as_offset(
+    offset_: FieldOffset,
+    field: common.Field,
+    /,
+) -> common.ConnectivityField:
+    offset_dim = np.squeeze(
+        np.where(list(map(lambda x: x == offset_.source, field.domain.dims)))
+    ).item()
+    new_connectivity = np.indices(field.ndarray.shape)[offset_dim] + field.ndarray
+    return common.connectivity(new_connectivity, codomain=offset_.source, domain=field.domain)
```

Comment:
I think this should raise `NotImplementedError`. this should never be reached if the dispatching is correct. This area of the code might change in the future, it's a bit confusing...